src/odemis/driver/xt_client.py

```diff
@@ -2358,6 +2367,73 @@ def _moveTo(self, future: futures.Future, pos: Dict[str, float], rel: bool = Fal
+
+                # The stage is not moving anymore, however in some rare cases, the server still reports the old
+                # position for a short while (typically < 1s). This can cause all sorts of confusion in the calling
+                # code, as it would seem that the move didn't have any effect, and later on, after the position
+                # is eventually updated that the stage has moved unexpectedly. So, wait until the reported position is
+                # (1) not identical to the starting position and (2) not too far from the target position.
+
+                # The stage is not moving anymore, but we still want to wait until the position has been updated
+                # TODO: base the timeout on the stage movement time estimation (est. *10)
+                timeout = 10  # s
+                expected_end_time = time.time() + timeout  # s
+                check_pos = True
+
+                # TODO: Include rotation and tilt to check stage movement to target position. Test it on a hardware.
+                # The below check to skip the check on stage movement to target position if it includes r ot t
+                # is temporary. It will be removed once the behavior is extended to all the axes and confirmed to work
+                # on a hardware.
+                if 'r' in pos or 't' in pos:
+                    logging.debug("Position requested to move includes t (rx/tilt) or r (rz/rotation). "
+                                  "Reported position accuracy is not checked.")
+                    check_pos = False
+
+                # Get the target position in absolute coordinates
+                if rel:
+                    target_pos = {ax: val + pos[ax] for ax, val in orig_pos.items()}
+                else:
+                    target_pos = pos
+
+                while check_pos:
+                    current_pos = self.parent.get_stage_position()
+                    # Every axis requested to move should have moved (compared to the position before starting)
+                    # if there is some change w.r.t starting position, proceed to check the target position
+
+                    # TODO Check all axes and confirm it works on the hardware
+                    # if all(current_pos[a] != op for a, op in orig_pos.items()):
+                    #     axes_updated = isNearPosition(current_pos=current_pos, target_position=target_pos,
+                    #                                   axes=set(orig_pos.keys()), atol_linear=1e-6,
+                    #                                   atol_rotation=numpy.radians(6))
+                    #     if axes_updated:
+                    #         logging.debug("Position has updated fully: from %s -> %s", orig_pos,
+                    #                       current_pos)
+                    #         break
+
+                    # TODO Remove the check only on x any y
+                    if all(current_pos[a] != op for a, op in orig_pos.items()):
```

Comment:
Avoid direct float comparison
```suggestion
                    if all(math.isclose(current_pos[a], op, abs_tol=1e-6) for a, op in orig_pos.items()):
```

Reply:
No. There is a very special reason for this: we want to check whether the value has changed. That's mostly all we care about. If it has changed, it means the xt-client has received an updated metadata. Maybe it's very nearby, because the stage actually hasn't really moved (due to hardware issue), but we now know the new position. 
So, on purpose it's not using float comparison. 

That deserves a comment though...